### DISCUSSION TITLE
Unable to select array elements of JSON request

### DISCUSSION THREAD

**Discussion by lauraled** [grafana/grafana-infinity-datasource#26] [Q&A] [answered]
Hi,

first of all, thank you for the development of this plugin, it's awesome!

However, I encountered an issue while trying to set up a JSON query and could use some help.

I have a API that returns data in JSON format. The returned data looks like this (simplified):
```
[
  {
    "numFound": 24,
    "limit": 100,
    "dataSet": "host",
    "items": [
       {
         "host_name": "Host1",
         "host_operating_system": "Windows 10"
       },
       {
         "host_name": "Host2",
	 "host_operating_system": "Windows 10"
       }
    ]
  }
]
```

I can access the properties "dataSet" or "limit" just fine by using these selectors:
![grafik](https://user-images.githubusercontent.com/42748335/105960234-39183b80-607d-11eb-862d-a9a7ac5ad81d.png)

However, I haven't managed to access the properties of the elements of the `items` array, which are the ones I need. I tried `items` and `items[*]` and `items.*` and probably a few others, but haven't had any luck yet. 
What am I doing wrong?

Thanks in advance.

**Comment by yesoreyeram** [comment, marked as answer]
You probably need `$.*.items` as your root element as shown below.

<img width="1118" alt="Screenshot 2021-01-27 at 09 38 19" src="https://user-images.githubusercontent.com/153843/105972476-8eefe200-6083-11eb-9d02-06bb9298e285.png">

**Reply by lauraled** [reply]
That works! Many thanks.

**Comment by GITVINOD** [comment]
Hi @lauraled 

I have a similar requirement, but here I should include "dataSet": "host" as well along with "items". Could you please suggest how to select this kind of request?

Probably, each dataset has multiple items

[
  {
    "numFound": 24,
    "limit": 100,
    "dataSet": "host",
    "items": [
       {
         "host_name": "Host1",
         "host_operating_system": "Windows 10"
       },
       {
         "host_name": "Host2",
	 "host_operating_system": "Windows 10"
       }
    ]
  }
{
    "numFound": 25,
    "limit": 100,
    "dataSet": "hostX",
    "items": [
       {
         "host_name": "Host1",
         "host_operating_system": "Windows 11"
       },
       {
         "host_name": "Host2",
	 "host_operating_system": "Windows 12"
       }
    ]
  }
]
